### PR TITLE
6 feat web UI basic design

### PR DESCRIPTION
**Pull request by takeshi8989**
# Description
issue #6 

- Componentize web ui
- redesign screens

### CODE REVIEW

File: matchya-web/src/components/AuthModal.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import FormInput from "./FormInput";
+import Button from "./Button";
+import matchIcon from "../assets/matchya-icon.png";
+import { useState } from "react";
```
Comment (kokiebisu) [inline matchya-web/src/components/AuthModal.tsx:4]: Let's sort the imports so it takes the order
1. Built-in libraries
2. Third-party libraries (ex.react)
3. Relative imports

File: matchya-web/src/components/AuthModal.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+interface Props {
```
Comment (kokiebisu) [inline matchya-web/src/components/AuthModal.tsx:6]: Let's name this AuthModalProps just so its easier to look up eventually

File: matchya-web/src/components/AuthModal.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  type: string // 'login' or 'signup'
+  action: (...args: string[]) => void // login or signup function
+  close: () => void
+  switchModal: () => void
+}
+
+const AuthModal = ({ type, action, close, switchModal }: Props) => {
+  const [companyName, setCompanyName] = useState('');
```
Comment (kokiebisu) [inline matchya-web/src/components/AuthModal.tsx:14]: Let's set the types for useState
```
const [companyName, setCompanyName] = useState<string>('');
```

File: matchya-web/src/components/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    text: string;
+    color?: string;
+    border?: boolean;
+    hover?: boolean;
+    outline?: boolean;
+    className?: string;
+    onClick?: (e?: React.MouseEvent) => void;
+}
+
+const Button = ({ text, color, border = true, hover = true, outline = true, className, onClick }: Props) => {
```
Comment (kokiebisu) [inline matchya-web/src/components/Button.tsx:11]: Let's have all the params with defaults at the end

File: matchya-web/src/components/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,51 @@
+interface Props {
```
Comment (kokiebisu) [inline matchya-web/src/components/Button.tsx:1]: Let's name this ButtonProps

File: matchya-web/src/components/Header.tsx [hunk 6]
```diff
@@ -1,21 +1,43 @@
 import { useState } from 'react'
 import matchyaIcon from '../assets/matchya-icon.png'
-import LoginModal from './LoginModal'
+import AuthModal from './AuthModal'
```
Comment (kokiebisu) [inline matchya-web/src/components/Header.tsx:3]: Let's organize imports

File: matchya-web/src/components/ScoreCard.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import { Score } from "../types"
+import { useState } from "react"
+
+
```
Comment (kokiebisu) [inline matchya-web/src/components/ScoreCard.tsx:4]: Don't need this extra line

File: matchya-web/src/components/ScoreCard.tsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+const ScoreCard = ({ score } : {score: Score}) => {
```
Comment (kokiebisu) [inline matchya-web/src/components/ScoreCard.tsx:5]: Let's create an interface for this named ScoreCardProps

File: matchya-web/src/types/index.ts [hunk 9]
```diff
@@ -0,0 +1,16 @@
+
```
Comment (kokiebisu) [inline matchya-web/src/types/index.ts:1]: Don't need these blank lines

File: matchya-web/src/types/index.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+export interface Score {
+    // id: string;
```
Comment (kokiebisu) [inline matchya-web/src/types/index.ts:4]: Let's remove this for now

File: matchya-web/src/types/index.ts [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+    name: string;
+    value: number;
+    details: string;
+    evaluations: Evaluation[];
+}
+
+export interface Evaluation {
+    // id: string;
```
Comment (kokiebisu) [inline matchya-web/src/types/index.ts:12]: Same as above

File: matchya-web/src/screens/Home.tsx [hunk 12]
```diff
@@ -1,51 +1,33 @@
+import Button from "../components/Button";
+import FormInput from "../components/FormInput";
```
Comment (kokiebisu) [inline matchya-web/src/screens/Home.tsx:2]: Organize imports

File: matchya-web/src/screens/Home.tsx [hunk 13]
```diff
@@ -1,51 +1,33 @@
+import Button from "../components/Button";
```
Comment (kokiebisu) [inline matchya-web/src/screens/Home.tsx:1]: Organize imporst

File: matchya-web/src/data/index.ts [hunk 14]
```diff
@@ -0,0 +1,25 @@
+import { Score } from '../types';
```
Comment (kokiebisu) [inline matchya-web/src/data/index.ts:1]: Let's name the folder 'mock'

File: matchya-web/src/data/index.ts [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+
+export const criteria: string[] = [
+    'Knows Python',
+    'Used Django before'
+];
+
+export const candidates: Score[] = [
```
Comment (kokiebisu) [inline matchya-web/src/data/index.ts:8]: Let's name the variables with the prefix of 'mock'
ex. mockCandidates, mockCriteria

File: matchya-web/src/screens/Settings.tsx [hunk 16]
```diff
@@ -59,4 +45,4 @@ const Settings = () => {
   );
 };
 
-export default Settings;
+export default Settings;
```
Comment (kokiebisu) [inline matchya-web/src/screens/Settings.tsx:48]: It's best practice to keep an empty line at the end of the file

File: matchya-web/src/screens/Home.tsx [hunk 17]
```diff
@@ -54,3 +36,62 @@ const Home = () => {
...
+          <FormInput label="GitHub Account URL" id="github_url" type="url" className="my-3" value={githubUrl} onChange={(e) => setGithubUrl(e.target.value)} />
+          <div className="flex justify-end w-full">
+            <Button text="Add" color="green" className="w-1/3" onClick={handleAddCandidate} /> 
+            <Button text="Cancel" color="red" className="w-1/3" onClick={close} />
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+}
```
Comment (kokiebisu) [inline matchya-web/src/screens/Home.tsx:97]: Best practice to add an empty line at the end of line

File: matchya-web/src/data/index.ts [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+    },
+    { 
+        name: 'Paul Carter', value: 6.0, 
+        details: 'Paul Carter\'s firm grasp of software development is reflected in his consistent problem-solving capabilities. His practical approach to complex challenges and reliable coding skills mark him as a promising candidate. His proactive learning attitude and ability to adapt to new technologies enhance his value as a team member.',
+        evaluations: [
+            { criteria: 'Knows Python', score: 6.0, reason: 'Paul has a competent grasp of Python, with experience in writing clean and maintainable code. He is comfortable with common Python libraries and frameworks and shows potential for further growth in this language.' },
+            { criteria: 'Used Django before', score: 6.0, reason: 'Paul has hands-on experience with Django, capable of handling typical web development tasks. He understands the framework\'s structure and conventions and can navigate its documentation to implement required functionality.' },
+        ]
+    },
+  ];
```
Comment (kokiebisu) [inline matchya-web/src/data/index.ts:25]: Empty line at the end of file

File: matchya-web/src/data/index.ts (hunk 14 above)
Comment (takeshi8989) [reply matchya-web/src/data/index.ts:1]: Should I rename it from types to mock? or should I rename the data folder to mock?

**Pull request merged**